templates/common/gcp/units/gcp-hostname.service.yaml

```
@@ -0,0 +1,21 @@
+name: gcp-hostname.service
+enabled: true
+contents: |
+  [Unit]
+  Description=Set GCP Transient Hostname
```

```suggestion
  Description=Set GCP Transient Hostname
  # Removal of this file signals firstboot completion
  ConditionPathExists=!/etc/ignition-machine-config-encapsulated.json
```
nitpick: service should run only after the node has pivoted